velox/vector/tests/VectorTest.cpp

```diff
@@ -1671,3 +1677,34 @@ TEST_F(VectorTest, multipleDictionariesOverLazy) {
     ASSERT_EQ(i, dict->as<SimpleVector<int32_t>>()->valueAt(i));
   }
 }
+
+/// Test lazy loading of nested dictionary vector
+TEST_F(VectorTest, selectiveLoadingOfLazyDictionaryNested) {
+  auto vectorMaker = std::make_unique<test::VectorMaker>(pool_.get());
+
+  vector_size_t size = 10;
+  auto indices =
+      makeIndices(size, [&](auto row) { return (row % 2 == 0) ? row : 0; });
+  auto data =
+      vectorMaker->flatVector<int32_t>(size, [](auto row) { return row; });
+  TestingLoader* loader = new TestingLoader(data);
```

Comment:
This pattern may leak memory. Let's use a smart pointer.

```
auto loader = std::make_unique<TestingLoader>(data);
auto loaderPtr = loader.get();

std::make_shared<LazyVector>(...std::move(loader));
```

Reply:
Thanks for catching this!